Add support for passing arbitrary C++ compiler flags. (#1526)
zeek/spicy: doc/programming/extending.rst

include paths, you can set the environment variable
``HILTI_CXX_INCLUDE_DIRS`` to a colon-separated list of additional
directories for ``spicyc`` to use when compiling C++ code.

Passing arbitrary compiler flags
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

If you need to set arbitrary compiler flags you can use the environment
variable ``HILTI_CXX_FLAGS``. If this value is set it will be added verbatim as

> last argument in the compiler invocation used when compiling C++ code.

Reviewer: ```suggestion
the last argument to the compiler invocation when compiling C++ code.
```
